src/views/CountryRiskWatch/HistoricalDataChart/index.tsx

```diff
@@ -0,0 +1,446 @@
+                    id: event.dtype.id,
+                    name: event.dtype.name,
+                };
+            },
+        ).filter(isDefined) ?? [],
+        (option) => option.id,
+    );
+
+    const filteredEvents = historicalDataResponse?.results?.filter(
+        (event) => {
+            if (!isValidDisaster(event.dtype)) {
+                return false;
+            }
+
+            if (isDefined(disasterFilter) && disasterFilter !== event.dtype.id) {
+                return false;
+            }
+
+            return true;
+        },
+    );
+
+    const maxValue = useMemo(
+        () => {
+            if (!filteredEvents) {
+                return 0;
+            }
+
+            const numAffectedList = filteredEvents.map(getNumAffected);
+            return Math.max(...numAffectedList);
+        },
+        [filteredEvents],
+    );
+
+    const yScale = useMemo(
+        () => {
+            const yBounds = getPrettyBreakableBounds(
+                { min: 0, max: maxValue },
+            );
+
+            return getScaleFunction(
+                yBounds,
+                { min: 0, max: chartBounds.height },
+                { start: chartMargin.top, end: chartMargin.bottom },
+                true,
+            );
+        },
+        [maxValue, chartBounds],
+    );
+
+    const xScale = useMemo(
+        () => (
+            getScaleFunction(
+                { min: 0, max: 12 },
+                { min: 0, max: chartBounds.width },
+                { start: chartMargin.left, end: chartMargin.right },
+            )
+        ),
+        [chartBounds],
+    );
+
+    const chartData = useMemo(
+        () => {
+            if (!filteredEvents) {
+                return undefined;
+            }
+
+            return filteredEvents.map(
+                (event) => {
+                    const disasterDate = new Date(event.disaster_start_date);
+                    const date = disasterDate.getDate();
+                    const month = disasterDate.getMonth();
+                    disasterDate.setMonth(month + 1);
+                    disasterDate.setDate(0);
+                    const totalDays = disasterDate.getDate();
+                    const monthValue = month + date / totalDays;
+
+                    const numAffected = getNumAffected(event);
+
+                    return {
+                        key: event.id,
+                        x: xScale(monthValue),
+                        y: yScale(numAffected),
+                        numAffected,
+                        disasterDate,
+                        disasterType: event.dtype.id,
+                        event,
+                    };
+                },
+            );
+        },
+        [filteredEvents, xScale, yScale],
+    );
+
+    const yAxisPoints = useMemo(
+        () => {
+            const numYAxisPoints = 6;
+            const diff = maxValue / (numYAxisPoints - 1);
+
+            return Array.from(Array(numYAxisPoints).keys()).map(
```

Comment:
Let's use Array.fill

Reply:
But why? Are we now using random methods?